### ISSUE TITLE
Horizontal scroll with big tables

### ISSUE THREAD

**Issue by IObregon** [koalyptus/TableFilter#440]
Good morning,

I have detected that when you have a big table (in my case with 12 th) and horizontal scrolling the place where the popup filters show can be changed.

![image](https://cloud.githubusercontent.com/assets/3036900/25334795/7eba835a-28f0-11e7-99be-7f44d888fae9.png)

![image](https://cloud.githubusercontent.com/assets/3036900/25334996/4962f362-28f1-11e7-89f0-f66cbaedf703.png)

![image](https://cloud.githubusercontent.com/assets/3036900/25334837/ae1ce8ae-28f0-11e7-8a8e-7a4fb25dd014.png)

This is worst when the popup filter opened is one of the lasts:

![image](https://cloud.githubusercontent.com/assets/3036900/25335056/9178670e-28f1-11e7-915e-504a02a12e92.png)

I think that this is because of the positioning of the popup when it is opened.

Is there any way to solve this?

**Comment by koalyptus** [comment]
Hi @IObregon,
Thanks for pointing this out, can you please post your configuration? Are you using the `responsive` option?

**Comment by IObregon** [comment]
Hi @koalyptus 
My configuration is a bit messy but :

```
var filtersConfig = {
					base_path: '../../js/lib/tablefilter/',
					state: {
				        types: ['local_storage'],
				        filters: true,
				        page_number: true,
				        page_length: true,
			      	},
					paging: true,
			        popup_filters: true,
// 			        responsive: true,
// 			        auto_filter: true,
// 			        auto_filter_delay: 1100,
			        results_per_page: [ [[#{lbl.paginacionjs.resultados}]], [10, 20, 50]],
					rows_counter: true,
			        btn_reset: true,
			        mark_active_columns: true,
			        highlight_keywords: true,
			        flt_cell_tag: 'th',
			        btn_first_page_html: '<a class="button button-red no-effect" href="javascript:void(0);"> <i class="fa fa-angle-double-left" aria-hidden="true"></i>',
			        btn_last_page_html: '<a class="button button-red no-effect" href="javascript:void(0);"> <i class="fa fa-angle-double-right" aria-hidden="true"></i>',
			        btn_next_page_html: '<a class="button button-red no-effect" href="javascript:void(0);"> <i class="fa fa-angle-right" aria-hidden="true"></i>',
			        btn_prev_page_html: '<a class="button button-red no-effect" href="javascript:void(0);"> <i class="fa fa-angle-left" aria-hidden="true"></i>',
			        popup_filters_image: '../../js/lib/tablefilter/style/themes/filter-1.png',
			        popup_filters_image_active: '../../js/lib/tablefilter/style/themes/filter-2.png',
			        paging_slc_css_class: 'select-orkli ocultarAzul',
			        results_slc_css_class: 'select-orkli',
			        toolbar_target_id: 'paginacion',
			        btn_reset_target_id: [[${reset}]],
			        page_text: [[#{lbl.paginacionjs.pagina}]],
			        of_text: [[#{lbl.paginacionjs.de}]],
			        rows_counter_text: [[${rowCounter}]],
			        help_instructions: true,
			        help_instructions_target_id: 'helpContainer',
			        help_instructions_html:[[#{msg.paginacionjs.ayuda}]],
			        thousands_separator: [[${numberFormatUtil.getThousandSeparator()}]],
			        decimal_separator: [[${numberFormatUtil.getDecimalSeparator()}]],
				};
```

I have tried with responsive at true too and the result is the same.

If you need more information please ask me.

**Comment by koalyptus** [comment]
Tx for posting your config, happy to see your advanced usage of the TableFilter 😄 
I am trying to understand what is triggering the horizontal scroll, a custom container element wrapping the HTML table or the `responsive` feature? From what I understand both scenarios have the same result on the popup filters located in the scrollable area. I will have a look.
Btw, be aware the `a` element is not closed in the paging buttons:
```javascript
btn_first_page_html: '<a class="button button-red no-effect" href="javascript:void(0);"> <i class="fa fa-angle-double-left" aria-hidden="true"></i></a>'
``` 
Cheers

**Comment by IObregon** [comment]
That's right. I forget every </a> but it worked perfectly.

I will give you some more clues:

1. My resolution is: 1366 x 768. 
2. The table have got 12 td's.
3. The first and the sixth have no filters.
4. This is some extra especific configuration for this table:
```
var extraConfig = {
						col_0 : 'none',
						col_6 : 'none',
						col_types: ['none', 'string',
						            'string',
						            'string',
						            'string',
						            'number',
						            'formatted-number',
						            { type: 'date', locale: [[${#locale.language} == 'eu' ? 'es' : ${#locale.language}]], format: ['{dd}-{months}-{yyyy}'] },
						            'none',
						            { type: 'date', locale: [[${#locale.language} == 'eu' ? 'es' : ${#locale.language}]], format: ['{dd}-{months}-{yyyy}'] }],
						extensions: [{ name: 'sort' }]
					};
```

Before creating the table filter I merge both objects like this:
```
jQuery.extend(filtersConfig, extraConfig);
				var tf = new TableFilter([[${tabla}]], filtersConfig);
				tf.init();
```

**Comment by koalyptus** [comment]
Tx for your detailed answers.
Just published a fix, starting from v0.5.9 the issue should be fixed, tell me how it goes.
Cheers

**Comment by IObregon** [comment]
It works perfectly. Thank you very much for your work.
